### PR TITLE
Fix multiple problems with Mobile month view

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#133289](http://central.tri.be/issues/133289) [C#133290](http://central.tri.be/issues/133290) [C#133291](http://central.tri.be/issues/133291) [C#133294](http://central.tri.be/issues/133294)
---

@lucatume when you get to code review this, please take ownership of merging this and fixing any possible problems since we will need these fixes for Alpha 2 and I will be out on Monday.

Leaving more info on the channel scheduled for Monday morning.

### CODE REVIEW

File: src/Tribe/Views/V2/Url.php [hunk 1]
```diff
@@ -246,4 +230,40 @@ public function get_query_arg_alias_of( $var, Context $context = null ) {
 	public function get_query_arg( $key, $default = null ) {
 		return Arr::get( (array) $this->get_query_args(), $key, $default );
 	}
+
+	/**
+	 * Returns all the aliases of the variable set in the Url query args, if any.
+	 *
+	 * @since 4.9.4
```
Comment (juanfra) [inline src/Tribe/Views/V2/Url.php]: If we're adding this one, let's set the `@since TBD`

File: src/Tribe/Views/V2/View.php [hunk 2]
```diff
@@ -918,8 +918,7 @@ public function is_publicly_visible() {
 	protected function setup_template_vars() {
 		$template_vars = [
 			'title'             => wp_title( null, false ),
-			'events'            => $this->repository->all(),
-			'url'               => $this->get_url( true ),
+			'events'            => $this->repository->all(), 'url'               => $this->get_url( true ),
```
Comment (juanfra) [inline src/Tribe/Views/V2/View.php]: Let's move this back to two lines.

File: src/Tribe/Views/V2/View.php [hunk 3]
```diff
@@ -1049,4 +1050,58 @@ public function get_today_url( $canonical = false ) {
...
+	 *
+	 * @since TBD
+	 *
+	 * @return string The date format, a valid PHP `date` function format, that should be used to build link labels.
+	 *
+	 * @see View::get_link_label(), the method using this method to build a link label.
+	 * @see date_i18n() as the formatted date will, then, be localized using this method.
+	 */
+	protected function get_label_format() {
+		return 'Y-m-d';
```
Comment (juanfra) [inline src/Tribe/Views/V2/View.php]: Maybe we want to add a filter here? Or at least leave a `@todo` to discuss it.

File: src/Tribe/Views/V2/Views/Month_View.php [hunk 4]
```diff
@@ -455,4 +365,59 @@ protected function calculate_grid_start_end( $date ) {
...
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @since TBD
+	 */
+	protected function get_label_format() {
+		// Something link "January".
+		return 'F';
```
Comment (juanfra) [inline src/Tribe/Views/V2/Views/Month_View.php]: Here as well. Maybe add a filter or at least a `@todo` to discuss it later.

File: src/Tribe/Views/V2/View.php (hunk 3 above)
Comment (lucatume) [reply src/Tribe/Views/V2/View.php]: Child classes are supposed to override this method.
We can add the filter for third-parties, but our code should override the method, not filter it.

File: src/Tribe/Views/V2/Views/Month_View.php (hunk 4 above)
Comment (lucatume) [reply src/Tribe/Views/V2/Views/Month_View.php]: I've resolved by filtering the label in the `View::get_link_label` method.

File: src/Tribe/Views/V2/View.php (hunk 3 above)
Comment (lucatume) [reply src/Tribe/Views/V2/View.php]: I've moved the filtering, per-view, in the `View::get_link_label` method.

**Pull request merged**